docs/fabric_mod_json_spec.md

```diff
@@ -0,0 +1,130 @@
+- **accessWidener**: A file path to an access widener relative to the mod root. If not present, assume the mod has no access widener.
+
+### Optional fields (dependency resolution)
+
+All of the following keys follow the format of a string->VersionRange dictionary, where the string key matches the desired ID.
+
+- **depends**: for these dependencies, a failure to match causes a hard failure,
+  - **recommends**: for these dependencies, a failure to match causes a soft failure (warning),
+  - **suggests**: these dependencies are not matched and are primarily used as metadata,
+  - **conflicts**: for these dependencies, a successful match causes a soft failure (warning),
+  - **breaks**: for these dependencies, a successful match causes a hard failure.
+
+### Optional fields (metadata)
+
+- **name**: Contains the user-facing mod name - a string value. If not present, assume it matches **id**.
+- **description**: Contains the user-facing mod description - a string value. If not present, assume empty string.
+- **authors**: Contains the direct authorship information - an array of Person values.
+- **contributors**: Contains the contributor information - an array of Person values.
+- **contact**: Contains the contact information for the project - a ContactInformation object.
+- **license**: Contains the licensing information - a string value, or an array of string values.
+  - This should provide the complete set of preferred licenses conveying the entire mod package. In other words, compliance with all listed licenses should be sufficient for usage, redistribution, etc. of the mod package as a whole.
+  - For cases where a part of code is dual-licensed, choose the preferred license. The list is not exhaustive, serves primarily as a kind of hint, and does not prevent you from granting additional rights/licenses on a case-by-case basis.
+  - To aid automated tools, it is recommended to use [SPDX License Identifiers](https://spdx.org/licenses/) for "open source" licenses.
+- **icon**: Contains the mod's icon, as a square .PNG file. (Minecraft resource packs use 128x128, but that is not a hard requirement - a power of two is, however, recommended.) Can be provided in one of two forms:
+  - A string, providing the path (from the mod's root) to a single .PNG file,
+  - A string->string dictionary, where the keys conform to widths of each PNG file, and the values are said files' paths.
+
+### Custom fields
+
+A **custom** field can be provided in the JSON. It is a dictionary of string keys to JSON elements, and the contents of said JSON elements are entirely arbitrary. The Fabric loader will make no effort to read its contents, aside from a situation in which custom features in older schema versions become official in future schema versions, as a compatibility measure - those will be adequately documented.
+
+It is recommended that the keys be namespaced with the ID of the mod relying on them (if such a mod exists) to prevent conflicts.
+
+## Version 0
```

Comment:
In my opinion v0 should have a seperate spec doc.

Reply:
Sure, this was just what asie had in the original. I'll remove this part.